src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -36,6 +37,26 @@ public static Index parseIndex(String oneBasedIndex) throws ParseException {
         return Index.fromOneBased(Integer.parseInt(trimmedIndex));
     }
 
+    /**
+     * Parses {@code oneBasedIndexes} into a {@code Set<Index>} and returns it.
+     * Leading and trailing whitespaces will be trimmed.
+     * @throws ParseException if the specified index is invalid (not non-zero unsigned integer) or there are duplicates.
+     */
+    public static Set<Index> parseIndexes(String oneBasedIndexes) throws ParseException {
+        String[] trimmedIndexArray = oneBasedIndexes.split("\\s+");
+        Set<Index> indexes = new HashSet<>();
+        for (String index : trimmedIndexArray) {
+            if (!StringUtil.isNonZeroUnsignedInteger(index)) {
+                throw new ParseException(MESSAGE_INVALID_INDEX);
+            }
+            if (!indexes.add(Index.fromOneBased(Integer.parseInt(index)))) {
```

Comment:
`Integer.parseInt(index)`  will throw `NumberFormatException` which will not be caught by `ParseException`

The inheritance is as such:

`ParseException -> IllegalValueException -> Exception`

`NumberFormatException -> IllegalArgumentException -> RuntimeException -> Exception`

Reply:
NumberFormatException is already handled by `StringUtil.isNonZeroUnsignedInteger(index)`